google/cloud/spanner_v1/_opentelemetry_tracing.py

```diff
@@ -105,4 +105,10 @@ def trace_call(name, session, extra_attributes=None, observability_options=None)
             span.record_exception(error)
             raise
         else:
-            span.set_status(Status(StatusCode.OK))
+            if span._status.status_code == StatusCode.UNSET:
```

Comment:
is span._status always not `None` or do we need a check for that?

Reply:
The opentelemetry package has this check
![image](https://github.com/user-attachments/assets/812fda13-681d-4302-be2a-17e131de751e)

Reply:
It is always defaulted to UNSET but sure, I can add that guard since we are poking the internals. Let me send up a quick change.

Reply:
I've added an extra check for `(not span._status) or `